pennylane/transforms/decompositions/clifford_t_transform.py

```diff
@@ -547,7 +547,7 @@ def circuit(x, y):
                 if not (op_param := op.simplify().data):
                     continue
                 # Decompose the RZ operation with a default wire
-                clifford_ops = _CLIFFORD_T_CACHE.query(qml.RZ(op_param[0], [0]))
+                clifford_ops = _CLIFFORD_T_CACHE.query(qml.RZ(op_param[0], [op.wires[0]]))
```

Comment:
This will be `[0]` to cache it independently of `wires`, as we are doing the mapping below.